src/util/git.js

```diff
@@ -44,32 +50,49 @@ export default class Git {
   hash: string;
   ref: string;
   cwd: string;
-  url: string;
+  gitUrl: GitUrl;
 
+  /**
+   * npm URLs contain a 'git+' scheme prefix, which is not understood by git.
+   * git "URLs" also allow an alternative scp-like syntax, so they're not standard URLs.
+   */
+  static npmUrlToGitUrl(npmUrl: string): GitUrl {
+    // Special case in npm, where ssh:// prefix is stripped to pass scp-like syntax
+    // which works as remote path only if there are no slashes before ':'
+    const match = npmUrl.match(/^git\+ssh:\/\/((?:[^@:\/]+@)?([^@:\/]+):.*)/);
```

Comment:
This is intentional. In scp host specifiers there is no password.

Git manual (`man git-clone`) specifies it as `[user@]host.xz:/~[user]/path/to/repo.git/`, although relative paths are also supported (scp man page documents the syntax as `[[user@]host1:]file`)